src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,86 @@
+package duke;
+
+import java.util.ArrayList;
+
+/**
+ * A class which scans the user input and decides which task to be carried out.
+ */
+public class Parser {
+    private String input;
+    private Ui ui;
+    private TaskList taskList;
+
+    /**
+     * Constructor of the class Parser.
+     *
+     * @param ui The UI used for Duke.
+     * @param taskList A list of tasks.
+     */
+    public Parser(Ui ui, TaskList taskList, String input) {
+        this.input = input;
+        this.ui = ui;
+        this.taskList = taskList;
+    }
+
+    /**
+     * Scans the user inputs and calls necessary methods based on the command input by the user.
+     */
+    public String run() {
```

Comment:
Is there a better way to "run" this method, seems kind long.
Perhaps you can make use of a dictionary to store commands. (Just a suggestion)
:)